app/src/main/java/com/example/myapplication/AdapterFactory.java

```diff
@@ -0,0 +1,18 @@
+package com.example.myapplication;
+
+import android.content.Context;
+
+import java.util.List;
+
+public class AdapterFactory {
```

Comment:
I created factory because you call twice in MainActivity Time_Adapter with same parameters